### PR TITLE
Handle invalid content type while reading input schema

### PR DESCRIPTION
**Pull request by atumbde**

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java [hunk 1]
```diff
@@ -145,7 +146,18 @@ public Representation post(Representation entity) {
   })
   private Representation uploadNewSchema()
       throws Exception {
-    File dataFile = getUploadContents();
+    File dataFile;
+    try {
+      dataFile = getUploadContents();
+    } catch (FileUploadBase.InvalidContentTypeException e) {
+      LOGGER.info("Invalid content type while adding new schema");
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: log any additional info here. URI?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java [hunk 2]
```diff
@@ -145,7 +146,18 @@ public Representation post(Representation entity) {
...
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST,
+          e.getMessage());
+    } catch (Exception e) {
+      LOGGER.error("Error reading request body while adding new schema", e);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: Log URI if possible

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java [hunk 3]
```diff
@@ -207,7 +219,18 @@ public Representation put(Representation entity) {
   private Representation uploadSchema(
       @Parameter(name = "schemaName", in = "path", description = "The name of the schema to get")
       String schemaName) throws Exception {
-    File dataFile = getUploadContents();
+    File dataFile;
+    try {
+      dataFile = getUploadContents();
+    } catch (FileUploadBase.InvalidContentTypeException e) {
+      LOGGER.info("Invalid content type while updating schema {}", schemaName);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: ditto

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java [hunk 4]
```diff
@@ -207,7 +219,18 @@ public Representation put(Representation entity) {
...
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST,
+          e.getMessage());
+    } catch (Exception e) {
+      LOGGER.error("Error reading request body while updating new schema: {}", schemaName, e);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: ditto, and can this code be merged to be in one place?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java (hunk 1 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: URIs are not useful. They just say /schema which we know already

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java (hunk 2 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: as above. URIs are not useful

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java (hunk 3 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: as above but this one has schema name which is logged (we don't have schema name above)

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java (hunk 4 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java]: Making separate method will require different exception to allow early return. It also needs if-else to handle optional schemaName in logs. So, there is no real benefit.

We can make a common method that handles schema update that is called by both PUT and POST apis. But there are subtle differences and no UTs. So, I want to limit the scope of this change.

**Pull request merged**